xds/src/main/java/io/grpc/xds/EdsLoadBalancer2.java

```diff
@@ -318,27 +269,7 @@ public void run() {
             if (lb == null) {
               lb = lbRegistry.getProvider(PRIORITY_POLICY_NAME).newLoadBalancer(lbHelper);
             }
-            if (localityPickingPolicy != null && endpointPickingPolicy != null) {
-              PriorityLbConfig config = generatePriorityLbConfig(cluster, edsServiceName,
-                  lrsServerName, localityPickingPolicy, endpointPickingPolicy, lbRegistry,
-                  prioritizedLocalityWeights);
-              // TODO(chengyuanzhang): to be deleted after migrating to use XdsClient API.
-              Attributes attributes;
-              if (lrsServerName != null) {
-                attributes =
-                    resolvedAddresses.getAttributes().toBuilder()
-                        .set(XdsAttributes.ATTR_CLUSTER_SERVICE_LOAD_STATS_STORE, loadStatsStore)
-                        .build();
-              } else {
-                attributes = resolvedAddresses.getAttributes();
-              }
-              lb.handleResolvedAddresses(
-                  resolvedAddresses.toBuilder()
-                      .setAddresses(endpointAddresses)
-                      .setAttributes(attributes)
-                      .setLoadBalancingPolicyConfig(config)
-                      .build());
-            }
+            handleResourceUpdate();
```

Comment:
Should check `resolvedAddresses  != null` first?

Reply:
The instantiation of `ChildLbState` is always followed by an invocation of `handleResolvedAddresses()`. So when `onChanged()` is being executed, `resolvedAddresses` will always have value.

Reply:
Well yes, it's invoked by `switchingLoadBalancer.handleResolvedAddresses()`. It's safe to check but that's alright.